### PR TITLE
Azure Managed Instance for Apache Cassandra (Cosmos DB)

### PR DESCRIPTION
**Pull request by TheovanKraay**

### CODE REVIEW

File: internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: networkValidate.SubnetID,
+			},
+
+			"node_count": {
+				Type:         pluginsdk.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntAtLeast(3),
```
Comment (katbyte) [inline internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go:71]: is there a maximum number here?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the pr @TheovanKraay - overall looks good aside from some comments i've left line. Additionally we need to  merge main in and fix the merge conflicts. otherwise provided tests pass this should be good to go!

File: internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+				Default:      3,
+			},
+			"sku_name": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+			"disk_capacity": {
```
[the remark this answers is not in the document]
Comment (TheovanKraay) [reply internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go:78]: it is number of P30 disks

[the remark this answers is not in the document]
Comment (TheovanKraay) [reply internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go:78]: ... to attach to each Cassandra node

**Comment by TheovanKraay** [comment]
> Thanks for the pr @TheovanKraay - overall looks good aside from some comments i've left line. Additionally we need to merge main in and fix the merge conflicts. otherwise provided tests pass this should be good to go!

@katbyte - apologies, I had problems with my local branch so I need to abandon this PR. The new PR is https://github.com/hashicorp/terraform-provider-azurerm/pull/14019. I believe I have addressed all your comments there.

**Pull request closed without merging**